cuckoo_hashing/cuckoo_hashing.h

```diff
@@ -58,6 +58,10 @@ class CuckooTable : public HashingTable {
 
   std::vector<uint64_t> AsRawVector() const final;
 
+  std::vector<uint64_t> AsRawVectorOfIds() const;
+
+  std::vector<bool> AsRawVectorOfOccupancy() const;
```

Comment:
maybe we could try to improve the naming of the functions. what do you think about `CopyElement/ItemIds`, and `CreateBinOccupancyVector` or `Show/GetBinOccupancy` (it though doesn't only show/get it...)? or do you have any better ideas? 

same about simple hashing

Reply:
Aren't they kind of getter functions, so naming in the style of `GetBinOccupancy` would be fitting?

Reply:
Not in the meaning I know. `Get*` is usually something that you would use to access a member variable in some way. Those functions would, on the other hand, do something and create new objects. So it's more like `ComputeBinOccupancy`, I would say.

Reply:
I'm fine with `ComputeBinOccupancy` and `ComputeBinEntryIds`. I could also replace `Compute` with `Obtain` since they don't do much computation but copy entry meta information to avoid confusion with `Get*`? What do you like more?

And should `AsRawVector` be renamed accordingly for consistency (would create compatibility issues)?

Reply:
`Obtain` or `Retrieve` sounds good, IMO. The first [seems to be](https://books.google.com/ngrams/graph?content=retrieve%2Cobtain&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Cretrieve%3B%2Cc0%3B.t1%3B%2Cobtain%3B%2Cc0#t1%3B%2Cretrieve%3B%2Cc0%3B.t1%3B%2Cobtain%3B%2Cc0) more commonly used.

Renaming `AsRawVector` also sounds like a good idea. Compatibility issues will be resolved quite quickly, I believe.

Reply:
i renamed them accordingly. Since the 1D functions ignore the bins (or assume bin = entry for cuckoo), I specified whether the function is concerned with bins or entries or both in the name.